include/pybind11/detail/class.h

```diff
@@ -87,6 +90,13 @@ inline PyTypeObject *make_static_property_type() {
         pybind11_fail("make_static_property_type(): failure in PyType_Ready()!");
     }
 
+#    if PY_VERSION_HEX >= 0x030C0000
+    // PRE 3.12 FEATURE FREEZE. PLEASE REVIEW AFTER FREEZE.
+    // Since Python-3.12 property-derived types are required to
+    // have dynamic attributes (to set `__doc__`)
+    enable_dynamic_attributes(heap_type);
+#    endif
```

Comment:
```suggestion
#endif
```

Not that it's great, but we should follow the current styling. Would be okay if it was updated (older versions of clang-format didn't handle it, and still not sure if newer versions indent based on the surrounding code).